packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -68,7 +67,8 @@ function CustomSelectControl( props: LegacyCustomSelectProps ) {
 					key={ key }
 					value={ name }
 					children={ __experimentalHint ? withHint : name }
-					{ ...rest }
+					style={ style }
+					className={ className }
```

Comment:
I will look into the forwarding of `...rest` and what issues that was causing, and I will double-check what the legacy component's behaviour was

Reply:
I can confirm that [the v1 `CustomSelectControl` only forwards `style` and `className` ](https://github.com/WordPress/gutenberg/blob/b76de86ffe89b437808a7a56c48021838fcb2d8a/packages/components/src/custom-select-control/index.js#L197-L209), and therefore this change seems correct.

I will make sure unit tests check this behaviour both for v1 and v2

Reply:
done